docs/articles/layouts.md

```diff
@@ -357,7 +359,99 @@ code (please note, that the value of a variable can be an expression):
 Parameterized fragments let developers create functional-like fragments
 that are easier to reuse. Read more about parameterized fragments in the
 Thymeleaf documentation: [Parameterizable fragment
-signatures](http://www.thymeleaf.org/doc/html/Using-Thymeleaf.html#parameterizable-fragment-signatures).
+signatures](http://www.thymeleaf.org/doc/tutorials/3.0/usingthymeleaf.html#parameterizable-fragment-signatures).
+
+### Fragment Expressions
+
+Thymeleaf 3.0 introduced a new type of expression as a part of the general Thymeleaf Standard Expression system: **Fragment Expressions**: 
+
+```xml
+    <div th:insert="~{fragments/footer :: footer}">...</div>
+```
+
+The idea of this syntax is to be able to use resolved fragments as any other kind of objects in the template execution context for later use:
+
+```xml
+<div th:replace="${#authentication.principal.isAdmin()} ? ~{fragments/footer :: footer-admin} : ~{fragments/footer :: footer-admin}">
+  &copy; 2016 The Static Templates
+</div>
+```
+
+Fragment expression allows creating fragments in a way such that they can be enriched with markup coming from the calling templates, resulting in a layout mechanism that is far more flexible than `th:insert` and `th:replace`only.
+
+#### Flexible Layout Example
+
+The `task/layout.html` file defines all the fragments that will be used by calling templates. The below `header` fragment takes `breadcrumb` parameter that will replace `ol` markup with its resolved value:
+
+```xml
+<!--/* Header fragment */-->
+<div th:fragment="header(breadcrumb)">
+    <ol class="breadcrumb container" th:replace="${breadcrumb}">
+        <li><a href="#">Home</a></li>
+    </ol>
+</div>
+```
+In the calling template (`task/task-list.html`) we will use a *Markup Selector* syntax to pass the element matching `.breadcrumb` selector:
+
+```xml
+<!--/* The markup with breadcrumb class will be passed to the header fragment */-->
+<header th:insert="task/layout :: header(~{ :: .breadcrumb})">
+    <ol class="breadcrumb container">
+        <li><a href="#">Home</a></li>
+        <li><a href="#" th:href="@{/task}">Tasks</a></li>
+    </ol>
+</header>
+```
+
+As a result, the following HTML will be generated for the `task/taks-list` view:
+
+```xml
+<header>
+    <ol class="breadcrumb container">
+        <li><a href="#">Home</a></li>
+        <li><a href="[...]">Tasks</a></li>
+    </ol>
+</header>
+```
```

Comment:
Actually, this code would not be the resulting one. The reason is, `th:insert` would insert the `<div>` tag itself inside the `<header>` (whereas the old `th:include` would not).

Options are either simply add the `<div>`to the result (which would be fine, I think, and my preferred option), or use the new `/content()` selector so that the expression ends up being:
```html
<header th:insert="task/layout :: header/content() (~{ :: .breadcrumb})">
...
```